diff --git a/configs/mx8mmea-ucom_defconfig b/configs/mx8mmea-ucom_defconfig
index 076f947bf2..139936f8d3 100644
--- a/configs/mx8mmea-ucom_defconfig
+++ b/configs/mx8mmea-ucom_defconfig
@@ -73,8 +73,8 @@ CONFIG_FASTBOOT_FLASH=y
 CONFIG_SYS_I2C_MXC_I2C1=y
 CONFIG_SYS_I2C_MXC_I2C2=y
 CONFIG_SYS_I2C_MXC_I2C3=y
-CONFIG_ENV_SIZE=0x2000
-CONFIG_ENV_OFFSET=0x1FE000
+CONFIG_ENV_SIZE=0x20000
+CONFIG_ENV_OFFSET=0x800000
 CONFIG_DM_GPIO=y
 CONFIG_DM_I2C=y
 CONFIG_SYS_I2C_MXC=y
@@ -133,3 +133,13 @@ CONFIG_VIDEO_ADV7535=y
 CONFIG_SYS_WHITE_ON_BLACK=y
 
 CONFIG_SPL_GZIP=y
+
+# Mender support
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
+CONFIG_ENV_OFFSET_REDUND=0x1000000
+
+# Improve boot time. The function ext4load takes about 40s to load
+# the kernel image (19 MB) without block cache being enabled
+CONFIG_BLOCK_CACHE=y
diff --git a/include/configs/mx8mmea-ucom.h b/include/configs/mx8mmea-ucom.h
index ff818bfb83..8df4045e6c 100644
--- a/include/configs/mx8mmea-ucom.h
+++ b/include/configs/mx8mmea-ucom.h
@@ -111,26 +111,26 @@
 	JAILHOUSE_ENV \
 	M_CORE_ENV \
 	"script=boot.scr\0" \
-	"image=Image\0" \
+	"image=boot/Image\0" \
 	"splashimage=0x50000000\0" \
 	"console=ttymxc1,115200\0" \
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=try\0" \
-	"fdt_file=imx8mm-ea-ucom-kit_v2.dtb\0" \
+	"fdt_file=boot/imx8mm-ea-ucom-kit_v2.dtb\0" \
 	"initrd_addr=0x43800000\0"		\
 	"initrd_high=0xffffffffffffffff\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
 	"mmcpart=" __stringify(CONFIG_SYS_MMC_IMG_LOAD_PART) "\0" \
 	"mmcroot=" CONFIG_MMCROOT " rootwait rw\0" \
 	"mmcautodetect=yes\0" \
-	"mmcargs=setenv bootargs ${jh_clk} console=${console} root=${mmcroot} " \
+	"mmcargs=setenv bootargs ${jh_clk} console=${console} root=${mender_kernel_root} " \
 		"${args_from_script}\0" \
 	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
-	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"loadimage=ext4load ${mender_uboot_root} ${loadaddr} ${image}\0" \
+	"loadfdt=ext4load ${mender_uboot_root} ${fdt_addr} ${fdt_file}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
@@ -166,14 +166,14 @@
 
 #define CONFIG_BOOTCOMMAND \
 	   "mmc dev ${mmcdev}; if mmc rescan; then " \
-		   "if run loadbootscript; then " \
-			   "run bootscript; " \
-		   "else " \
+			   "run mender_setup; " \
 			   "if run loadimage; then " \
 				   "run mmcboot; " \
-			   "else run netboot; " \
+				   "run mender_try_to_recover; " \
+			   "else " \
+				   "run mender_try_to_recover; " \
+				   "run netboot; " \
 			   "fi; " \
-		   "fi; " \
 	   "else booti ${loadaddr} - ${fdt_addr}; fi"
 #endif
 
@@ -191,7 +191,7 @@
 
 #define CONFIG_ENV_OVERWRITE
 #define CONFIG_SYS_MMC_ENV_DEV		2   /* USDHC2 */
-#define CONFIG_SYS_MMC_ENV_PART		1   /* 0=user area, 1=1st MMC boot part., 2=2nd MMC boot part. */
+#define CONFIG_SYS_MMC_ENV_PART		0   /* 0=user area, 1=1st MMC boot part., 2=2nd MMC boot part. */
 #define CONFIG_MMCROOT			"/dev/mmcblk2p2"  /* USDHC2 */
 
 /* Size of malloc() pool */
