diff --git a/configs/mx8mqea-com_defconfig b/configs/mx8mqea-com_defconfig
index 948e3ed..d19b987 100644
--- a/configs/mx8mqea-com_defconfig
+++ b/configs/mx8mqea-com_defconfig
@@ -72,3 +72,12 @@ CONFIG_USB_DWC3=y
 CONFIG_USB_DWC3_GADGET=y
 CONFIG_DM_PMIC_BD71837=y
 CONFIG_SPL_GZIP=y
+
+# Mender support
+CONFIG_BOOTCOUNT_LIMIT=y
+CONFIG_BOOTCOUNT_ENV=y
+
+# Improve boot time. The function ext4load takes about 40s to load
+# the kernel image (19 MB) without block cache being enabled
+CONFIG_BLOCK_CACHE=y
+
diff --git a/include/configs/mx8mqea-com.h b/include/configs/mx8mqea-com.h
index 45a895e..f6514f0 100644
--- a/include/configs/mx8mqea-com.h
+++ b/include/configs/mx8mqea-com.h
@@ -125,12 +125,12 @@
 	CONFIG_MFG_ENV_SETTINGS \
 	JAILHOUSE_ENV \
 	"script=boot.scr\0" \
-	"image=Image\0" \
+	"image=boot/Image\0" \
 	"console=ttymxc0,115200 earlycon=ec_imx6q,0x30860000,115200\0" \
 	"fdt_addr=0x43000000\0"			\
 	"fdt_high=0xffffffffffffffff\0"		\
 	"boot_fdt=try\0" \
-	"fdt_file=fsl-imx8mq-ea-com-kit_v2.dtb\0" \
+	"fdt_file=boot/fsl-imx8mq-ea-com-kit_v2.dtb\0" \
 	"initrd_addr=0x43800000\0"		\
 	"initrd_high=0xffffffffffffffff\0" \
 	"mmcdev="__stringify(CONFIG_SYS_MMC_ENV_DEV)"\0" \
@@ -139,12 +139,12 @@
 	"mmcautodetect=yes\0" \
 	"fmac_txrx_opt=brcmfmac.sdio_wq_highpri=1\0" \
 	"mmcargs=setenv bootargs ${jh_clk} console=${console} "\
-		"root=${mmcroot} ${fmac_txrx_opt} ${args_from_script}\0 " \
+		"root=${mender_kernel_root} ${fmac_txrx_opt} ${args_from_script}\0 " \
 	"loadbootscript=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${script};\0" \
 	"bootscript=echo Running bootscript from mmc ...; " \
 		"source\0" \
-	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
-	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}\0" \
+	"loadimage=ext4load ${mender_uboot_root} ${loadaddr} ${image}\0" \
+	"loadfdt=ext4load ${mender_uboot_root} ${fdt_addr} ${fdt_file}\0" \
 	"mmcboot=echo Booting from mmc ...; " \
 		"run mmcargs; " \
 		"if test ${boot_fdt} = yes || test ${boot_fdt} = try; then " \
@@ -182,9 +182,13 @@
 		   "if run loadbootscript; then " \
 			   "run bootscript; " \
 		   "else " \
+			   "run mender_setup; " \
 			   "if run loadimage; then " \
 				   "run mmcboot; " \
-			   "else run netboot; " \
+				   "run mender_try_to_recover; " \
+			   "else " \
+				   "run mender_try_to_recover; " \
+				   "run netboot; " \
 			   "fi; " \
 		   "fi; " \
 	   "else booti ${loadaddr} - ${fdt_addr}; fi"
@@ -202,10 +206,11 @@
         (CONFIG_SYS_INIT_RAM_ADDR + CONFIG_SYS_INIT_SP_OFFSET)
 
 #define CONFIG_ENV_OVERWRITE
-#define CONFIG_ENV_OFFSET		(SZ_2M - CONFIG_ENV_SIZE)
-#define CONFIG_ENV_SIZE			SZ_8K
+/* Use mender default instead */
+/*#define CONFIG_ENV_OFFSET		(SZ_2M - CONFIG_ENV_SIZE)*/
+/*#define CONFIG_ENV_SIZE			SZ_8K*/
 #define CONFIG_SYS_MMC_ENV_DEV		0   /* USDHC1 */
-#define CONFIG_SYS_MMC_ENV_PART		1   /* 0=user area, 1=1st MMC boot part., 2=2nd MMC boot part. */
+#define CONFIG_SYS_MMC_ENV_PART		0   /* 0=user area, 1=1st MMC boot part., 2=2nd MMC boot part. */
 #define CONFIG_MMCROOT			"/dev/mmcblk0p2"  /* USDHC1 */
 
 /* Size of malloc() pool */
